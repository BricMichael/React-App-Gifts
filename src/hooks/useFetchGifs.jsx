import { useState, useEffect } from 'react';
import { getGifs } from '../Helpers/Get_Gifs';


export const useFetchGifs = ( category ) => {

    const [ object, setObject] = useState({
        data: [],
        loading: true,
    })

    useEffect( () => {
        getGifs( category )
        .then( imgs => {

            setObject({
                data: imgs,
                loading: false

            });

        })

    }, [ category ])

    return object;

}