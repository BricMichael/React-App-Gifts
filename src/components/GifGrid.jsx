import React from 'react';
import GifItem from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs'
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );
   
    return(
        <>
            <h2 className = 'color_h2'>{ category }</h2>
            { loading && <p className="card animate__animated animate__flash">Loading</p> }
            <div className = 'card-grip'>
                {

                    images.map( imgs => (

                        <GifItem 
                          key = { imgs.id } 
                          { ...imgs } 
                        />

                    ))
                }   
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;