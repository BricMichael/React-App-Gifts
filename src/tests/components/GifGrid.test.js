import React from 'react';
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el componente <GifGrid />', () => {

    

    test('Fotografiar el componente', () => {
       

       useFetchGifs.mockReturnValue({
           data: [],
           loading: true
       })

       const wrapper = shallow( <GifGrid category='Perfect' />)
       expect( wrapper ).toMatchSnapshot();
       
    })

    test('Asegurarse de que existe el componente', () => {
        
        const gifs = [{
            url: 'hhtps/images/pngAndjpg.png',
            title: 'user',
            id: 'ABC'
        },{
            url: 'hhtps/images/pngAndjpg.png',
            title: 'user',
            id: 'wxc'
          }
        ]

        
   
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow( <GifGrid category='Perfect' />)

        expect( wrapper ).toMatchSnapshot()

        expect( wrapper.find('p').exists()).toBe( false ) // no deberia haber parrafos 
 
        const component = wrapper.find('GifItem').length;  // accediendo a un componente y su longitud
        expect( component ).toBe( gifs.length )
    })

    

})