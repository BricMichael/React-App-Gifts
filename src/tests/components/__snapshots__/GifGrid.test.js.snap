// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Asegurarse de que existe el componente 1`] = `
<Fragment>
  <h2
    className="color_h2"
  >
    Perfect
  </h2>
  <div
    className="card-grip"
  >
    <GifItem
      id="ABC"
      key="ABC"
      title="user"
      url="hhtps/images/pngAndjpg.png"
    />
    <GifItem
      id="wxc"
      key="wxc"
      title="user"
      url="hhtps/images/pngAndjpg.png"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> Fotografiar el componente 1`] = `
<Fragment>
  <h2
    className="color_h2"
  >
    Perfect
  </h2>
  <p
    className="card animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grip"
  />
</Fragment>
`;
